Sources/Datadog/Core/Upload/DataUploadConditions.swift

```diff
@@ -16,34 +40,52 @@ internal struct DataUploadConditions {
     let batteryStatus: BatteryStatusProviderType?
     let networkConnectionInfo: NetworkConnectionInfoProviderType
 
-    func canPerformUpload() -> Bool {
+    func canPerformUpload() -> Report {
```

Comment:
nit: the `Report` doesn't sound well to me. I'd expect kind of `Result` or maybe even:
```swift
func currentConditions() -> Conditions

enum Conditions {
   case sufficient
   case insufficient(blockers: [Blocker])
}
```
But as it's a naming thing - feel free to keep it your way 👍.

Reply:
I have considered naming this `Conditions` as well but couldn't find a way to have repeating and maybe confusing pattern like `DataUploadConditions` returns some other kind of `Conditions`. That being said, your proposal still sounds clearer when used at call-sites.